e2e/dragonfly_controller_test.go

```diff
@@ -472,6 +472,18 @@ var _ = Describe("Dragonfly Lifecycle tests", Ordered, func() {
 			// check for affinity
 			Expect(ss.Spec.Template.Spec.Affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution).To(Equal(newAffinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution))
 		})
+
+		It("Cleanup", func() {
+			var df dragonflydbiov1alpha1.Dragonfly
+			err := k8sClient.Get(ctx, types.NamespacedName{
+				Name:      name,
+				Namespace: namespace,
+			}, &df)
+			Expect(err).To(BeNil())
+
+			err = k8sClient.Delete(ctx, &df)
```

Comment:
Should we also check if the related resources are also delete (i.e. deployments, service etc)?

Reply:
Deletion is not the purview of the Operator. It is done automatically by the garbage collector in k8s. This is through the Owner reference that we pass to resources.

Reply:
Agreed